test/blackbox-tests/test-cases/promote/non-existent-empty.t/run.t

```diff
@@ -7,9 +7,11 @@ Tests for promoting with non existent reference
   [1]
 
   $ dune build @blah-non-existent
+  File "x-non-existent", line 1, characters 0-0:
+  Error: Files _build/default/x-non-existent and _build/default/x.gen differ.
```

Comment:
This error message should be clearer when one of the files doesn't exist.

Also, can you add a test case where we run a real diff program? I'm pretty sure this is going to give an ugly error that we're passing wrong arguments to diff at the moment. We should still be passing /dev/null to diff in such cases I imagine.